-1

Tested at: 09a73891adef217340e35a30cf80e23c6a1a3423

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ab57a7/1155/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found compilation error when building: 

<pre>>> Building shared library tmp/slc7_amd64_gcc700/src/L1TriggerConfig/L1GtConfigProducers/src/L1TriggerConfigL1GtConfigProducers/libL1TriggerConfigL1GtConfigProducers.so
tmp/slc7_amd64_gcc700/src/L1TriggerConfig/L1GtConfigProducers/src/L1TriggerConfigL1GtConfigProducers/L1GtTriggerMenuXmlParser.cc.o: In function `L1GtTriggerMenuXmlParser::initXML(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
L1GtTriggerMenuXmlParser.cc:(.text+0x2515): undefined reference to `cms::concurrency::xercesInitialize()'
tmp/slc7_amd64_gcc700/src/L1TriggerConfig/L1GtConfigProducers/src/L1TriggerConfigL1GtConfigProducers/L1GtTriggerMenuXmlParser.cc.o: In function `L1GtTriggerMenuXmlParser::cleanupXML(xercesc_3_1::XercesDOMParser*)':
L1GtTriggerMenuXmlParser.cc:(.text+0x1db): undefined reference to `cms::concurrency::xercesTerminate()'
collect2: error: ld returned 1 exit status
gmake: *** [tmp/slc7_amd64_gcc700/src/L1TriggerConfig/L1GtConfigProducers/src/L1TriggerConfigL1GtConfigProducers/libL1TriggerConfigL1GtConfigProducers.so] Error 1
Leaving library rule at L1TriggerConfig/L1GtConfigProducers
>> Leaving Package L1TriggerConfig/L1GtConfigProducers
>> Package L1TriggerConfig/L1GtConfigProducers built
>> Leaving Package CondCore/Utilities arg1=
</pre>

--- a/FWCore/Catalog/BuildFile.xml
+++ b/FWCore/Catalog/BuildFile.xml
@@ -1,7 +1,6 @@
 <use   name="FWCore/ServiceRegistry"/>
 <use   name="FWCore/Utilities"/>
-<use   name="Utilities/Xerces"/>
-<use   name="xerces-c"/>
+<use   name="tinyxml2"/>
 <use   name="boost"/>
 <export>
   <lib   name="1"/>

--- a/FWCore/Catalog/interface/FileLocator.h
+++ b/FWCore/Catalog/interface/FileLocator.h
@@ -6,7 +6,7 @@
 #include <map>
 #include <utility>
 #include <regex>
-#include <xercesc/dom/DOM.hpp>
+#include "tinyxml2.h"
 
 namespace edm {
 
@@ -36,7 +36,7 @@ namespace edm {
 
     void init(std::string const& catUrl, bool fallback);
 
-    void parseRule(xercesc::DOMNode* ruleNode, ProtocolRules& rules);
+    void parseRule(tinyxml2::XMLElement* ruleNode, ProtocolRules& rules);
 
     std::string applyRules(ProtocolRules const& protocolRules,
                            std::string const& protocol,

--- a/FWCore/Catalog/src/FileLocator.cc
+++ b/FWCore/Catalog/src/FileLocator.cc
@@ -2,10 +2,6 @@
 #include "FWCore/Catalog/interface/SiteLocalConfig.h"
 #include "FWCore/ServiceRegistry/interface/Service.h"
 
-#include <xercesc/parsers/XercesDOMParser.hpp>
-#include "Utilities/Xerces/interface/Xerces.h"
-#include "Utilities/Xerces/interface/XercesStrUtils.h"
-
 #include <boost/algorithm/string.hpp>
 #include <boost/algorithm/string/replace.hpp>
 
@@ -14,9 +10,6 @@
 #include <fstream>
 #include <iostream>
 #include <sstream>
-
-using namespace xercesc;
-using namespace cms::xerces;
 
 namespace {
 
@@ -36,28 +29,21 @@ namespace {
     // std::cerr << "Final string: " << result << std::endl;
     return result;
   }
+
+  constexpr char const* const kEmptyString = "";
+
+  const char* safe(const char* iCheck) {
+    if (iCheck == nullptr) {
+      return kEmptyString;
+    }
+    return iCheck;
+  }
+
 }  // namespace
 
 namespace edm {
 
-  int FileLocator::s_numberOfInstances = 0;
-
   FileLocator::FileLocator(std::string const& catUrl, bool fallback) : m_destination("any") {
-    try {
-      //  << "Xerces-c initialization Number "
-      //   << s_numberOfInstances <<
-      if (s_numberOfInstances == 0) {
-        cms::concurrency::xercesInitialize();
-      }
-    } catch (XMLException const& e) {
-      // << "Xerces-c error in initialization \n"
-      //      << "Exception message is:  \n"
-      //      << toString(e.getMessage()) <<
-      throw cms::Exception("TrivialFileCatalog",
-                           std::string("Fatal Error on edm::FileLocator:") + toString(e.getMessage()));
-    }
-    ++s_numberOfInstances;
-
     init(catUrl, fallback);
 
     // std::cout << m_protocols.size() << " protocols" << std::endl;
@@ -81,32 +67,27 @@ namespace edm {
     return out;
   }
 
-  void FileLocator::parseRule(DOMNode* ruleNode, ProtocolRules& rules) {
-    if (!ruleNode) {
+  void FileLocator::parseRule(tinyxml2::XMLElement* ruleElement, ProtocolRules& rules) {
+    if (!ruleElement) {
       throw cms::Exception("TrivialFileCatalog", std::string("TrivialFileCatalog::connect: Malformed trivial catalog"));
     }
 
-    // ruleNode is actually always a DOMElement because it's the result of
-    // a `getElementsByTagName()` in the calling method.
-    DOMElement* ruleElement = static_cast<DOMElement*>(ruleNode);
-
-    std::string const protocol = toString(ruleElement->getAttribute(uStr("protocol").ptr()));
-    std::string destinationMatchRegexp = toString(ruleElement->getAttribute(uStr("destination-match").ptr()));
-
-    if (destinationMatchRegexp.empty()) {
+    auto const protocol = safe(ruleElement->Attribute("protocol"));
+    auto destinationMatchRegexp = ruleElement->Attribute("destination-match");
+    if (destinationMatchRegexp == nullptr or destinationMatchRegexp[0] == 0) {
       destinationMatchRegexp = ".*";
     }
 
-    std::string const pathMatchRegexp = toString(ruleElement->getAttribute(uStr("path-match").ptr()));
-    std::string const result = toString(ruleElement->getAttribute(uStr("result").ptr()));
-    std::string const chain = toString(ruleElement->getAttribute(uStr("chain").ptr()));
+    auto const pathMatchRegexp = safe(ruleElement->Attribute("path-match"));
+    auto const result = safe(ruleElement->Attribute("result"));
+    auto const chain = safe(ruleElement->Attribute("chain"));
 
     Rule rule;
     rule.pathMatch.assign(pathMatchRegexp);
     rule.destinationMatch.assign(destinationMatchRegexp);
     rule.result = result;
     rule.chain = chain;
-    rules[protocol].push_back(rule);
+    rules[protocol].emplace_back(std::move(rule));
   }
 
   void FileLocator::init(std::string const& catUrl, bool fallback) {
@@ -177,15 +158,14 @@ namespace edm {
 
     configFile.close();
 
-    auto parser = std::make_unique<XercesDOMParser>();
-    try {
-      parser->setValidationScheme(XercesDOMParser::Val_Auto);
-      parser->setDoNamespaces(false);
-      parser->parse(m_filename.c_str());
-      DOMDocument* doc = parser->getDocument();
-      assert(doc);
+    tinyxml2::XMLDocument doc;
+    auto loadErr = doc.LoadFile(m_filename.c_str());
+    if (loadErr != tinyxml2::XML_SUCCESS) {
+      throw cms::Exception("TrivialFileCatalog")
+          << "tinyxml file load failed with error : " << doc.ErrorStr() << std::endl;
+    }
 
-      /* trivialFileCatalog matches the following xml schema
+    /* trivialFileCatalog matches the following xml schema
 	 FIXME: write a proper DTD
 	 <storage-mapping>
 	 <lfn-to-pfn protocol="direct" destination-match=".*"
@@ -197,31 +177,17 @@ namespace edm {
 	 </storage-mapping>
       */
 
-      /*first of all do the lfn-to-pfn bit*/
-      {
-        DOMNodeList* rules = doc->getElementsByTagName(uStr("lfn-to-pfn").ptr());
-        XMLSize_t const ruleTagsNum = rules->getLength();
+    auto rootElement = doc.RootElement();
+    /*first of all do the lfn-to-pfn bit*/
+    for (auto el = rootElement->FirstChildElement("lfn-to-pfn"); el != nullptr;
+         el = el->NextSiblingElement("lfn-to-pfn")) {
+      parseRule(el, m_directRules);
+    }
 
-        // FIXME: we should probably use a DTD for checking validity
-
-        for (XMLSize_t i = 0; i < ruleTagsNum; ++i) {
-          DOMNode* ruleNode = rules->item(i);
-          parseRule(ruleNode, m_directRules);
-        }
-      }
-      /*Then we handle the pfn-to-lfn bit*/
-      {
-        DOMNodeList* rules = doc->getElementsByTagName(uStr("pfn-to-lfn").ptr());
-        XMLSize_t ruleTagsNum = rules->getLength();
-
-        for (XMLSize_t i = 0; i < ruleTagsNum; ++i) {
-          DOMNode* ruleNode = rules->item(i);
-          parseRule(ruleNode, m_inverseRules);
-        }
-      }
-    } catch (xercesc::DOMException const& e) {
-      throw cms::Exception("TrivialFileCatalog")
-          << "Xerces XML parser threw this exception: " << cStr(e.getMessage()).ptr() << std::endl;
+    /*Then we handle the pfn-to-lfn bit*/
+    for (auto el = rootElement->FirstChildElement("pfn-to-lfn"); el != nullptr;
+         el = el->NextSiblingElement("pfn-to-lfn")) {
+      parseRule(el, m_inverseRules);
     }
   }
 

--- a/FWCore/Catalog/test/BuildFile.xml
+++ b/FWCore/Catalog/test/BuildFile.xml
@@ -1,6 +1,5 @@
 <use   name="FWCore/Catalog"/>
-<use   name="FWCore/PluginManager"/>
 <use   name="FWCore/ServiceRegistry"/>
-<use   name="FWCore/ParameterSetReader"/>
+<use   name="catch2"/>
 <bin   name="edmCatalogFileLocator_t" file="FileLocator_t.cpp">
 </bin>

--- a/FWCore/Catalog/test/FileLocator_t.cpp
+++ b/FWCore/Catalog/test/FileLocator_t.cpp
@@ -1,93 +1,113 @@
 #include "FWCore/Catalog/interface/FileLocator.h"
+#include "FWCore/Catalog/interface/SiteLocalConfig.h"
 #include "FWCore/Utilities/interface/Exception.h"
 #include "FWCore/ServiceRegistry/interface/ServiceRegistry.h"
-#include "FWCore/PluginManager/interface/standard.h"
-#include "FWCore/PluginManager/interface/PluginManager.h"
-#include "FWCore/ParameterSetReader/interface/ParameterSetReader.h"
 #include <boost/filesystem.hpp>
 #include <iostream>
 
-int main() {
-  try {
-    edmplugin::PluginManager::configure(edmplugin::standard::config());
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
 
-    std::string config =
-        "import FWCore.ParameterSet.Config as cms\n"
-        "process = cms.Process('edmFileUtil')\n"
-        "process.InitRootHandlers = cms.Service('InitRootHandlers')\n"
-        "process.SiteLocalConfigService = cms.Service('SiteLocalConfigService')\n";
+namespace {
+  class TestSiteLocalConfig : public edm::SiteLocalConfig {
+  public:
+    TestSiteLocalConfig(std::string catalog) : m_catalog(std::move(catalog)) {}
+    std::string const dataCatalog(void) const final { return m_catalog; }
+    std::string const fallbackDataCatalog(void) const final { return std::string(); }
+    std::string const lookupCalibConnect(std::string const& input) const final { return std::string(); }
+    std::string const rfioType(void) const final { return std::string(); }
 
-    //create the services
-    std::unique_ptr<edm::ParameterSet> params;
-    edm::makeParameterSets(config, params);
-    edm::ServiceToken tempToken(edm::ServiceRegistry::createServicesFromConfig(std::move(params)));
+    std::string const* sourceCacheTempDir() const final { return nullptr; }
+    double const* sourceCacheMinFree() const final { return nullptr; }
+    std::string const* sourceCacheHint() const final { return nullptr; }
+    std::string const* sourceCloneCacheHint() const final { return nullptr; }
+    std::string const* sourceReadHint() const final { return nullptr; }
+    unsigned int const* sourceTTreeCacheSize() const final { return nullptr; }
+    unsigned int const* sourceTimeout() const final { return nullptr; }
+    bool enablePrefetching() const final { return false; }
+    unsigned int debugLevel() const final { return 0; }
+    std::vector<std::string> const* sourceNativeProtocols() const final { return nullptr; }
+    struct addrinfo const* statisticsDestination() const final {
+      return nullptr;
+    }
+    std::set<std::string> const* statisticsInfo() const final { return nullptr; }
+    std::string const& siteName(void) const final {
+      static const std::string s_value = "TEST";
+      return s_value;
+    }
 
-    //make the services available
+  private:
+    std::string m_catalog;
+  };
+}  // namespace
+
+TEST_CASE("FileLocator", "[filelocator]") {
+  std::string CMSSW_BASE(std::getenv("CMSSW_BASE"));
+  std::string CMSSW_RELEASE_BASE(std::getenv("CMSSW_RELEASE_BASE"));
+  std::string file_name("/src/FWCore/Catalog/test/simple_catalog.xml");
+  std::string full_file_name = boost::filesystem::exists((CMSSW_BASE + file_name).c_str())
+                                   ? CMSSW_BASE + file_name
+                                   : CMSSW_RELEASE_BASE + file_name;
+
+  //create the services
+  edm::ServiceToken tempToken(edm::ServiceRegistry::createContaining(std::unique_ptr<edm::SiteLocalConfig>(
+      new TestSiteLocalConfig(std::string("trivialcatalog_file:") + full_file_name + "?protocol=xrd"))));
+
+  //make the services available
+  SECTION("standard") {
+    edm::ServiceRegistry::Operate operate(tempToken);
+    edm::FileLocator fl("", false);
+
+    const std::array<const char*, 7> lfn = {{"/bha/bho",
+                                             "bha",
+                                             "file:bha",
+                                             "file:/bha/bho",
+                                             "/castor/cern.ch/cms/bha/bho",
+                                             "rfio:/castor/cern.ch/cms/bha/bho",
+                                             "rfio:/bha/bho"}};
+
+    CHECK("/storage/path/store/group/bha/bho" == fl.pfn("/store/group/bha/bho"));
+    for (auto file : lfn) {
+      CHECK("" == fl.pfn(file));
+    }
+
+    CHECK(fl.lfn("/storage/path/store/group/bha/bho") == "/store/group/bha/bho");
+    CHECK(fl.lfn("/store/group/bha/bho") == "");
+    for (auto file : lfn) {
+      CHECK("" == fl.lfn(file));
+    }
+  }
+
+  SECTION("override") {
     edm::ServiceRegistry::Operate operate(tempToken);
 
-    {
-      edm::FileLocator fl("", false);
+    std::string file_name("/src/FWCore/Catalog/test/override_catalog.xml");
+    std::string full_file_name = boost::filesystem::exists((CMSSW_BASE + file_name).c_str())
+                                     ? CMSSW_BASE + file_name
+                                     : CMSSW_RELEASE_BASE + file_name;
 
-      const char* lfn[] = {"/store/group/bha/bho",
-                           "/bha/bho",
-                           "bha",
-                           "file:bha",
-                           "file:/bha/bho",
-                           "/castor/cern.ch/cms/bha/bho",
-                           "rfio:/castor/cern.ch/cms/bha/bho",
-                           "rfio:/bha/bho"};
-      int nfile = 8;
+    edm::FileLocator fl(("trivialcatalog_file:" + full_file_name + "?protocol=override").c_str(), false);
 
-      std::cout << "lfn2pfn" << std::endl;
-      for (int i = 0; i < nfile; ++i)
-        std::cout << lfn[i] << " -> " << fl.pfn(lfn[i]) << std::endl;
+    std::array<const char*, 8> lfn = {{"/store/group/bha/bho",
+                                       "/bha/bho",
+                                       "bha",
+                                       "file:bha",
+                                       "file:/bha/bho",
+                                       "/castor/cern.ch/cms/bha/bho",
+                                       "rfio:/castor/cern.ch/cms/bha/bho",
+                                       "rfio:/bha/bho"}};
 
-      std::cout << "pfn2lfn" << std::endl;
-      for (int i = 0; i < nfile; ++i)
-        std::cout << lfn[i] << " -> " << fl.lfn(lfn[i]) << std::endl;
+    auto const overriden_file =
+        "/store/unmerged/relval/CMSSW_3_8_0_pre3/RelValZTT/GEN-SIM-DIGI-RAW-HLTDEBUG/START38_V2-v1/0666/"
+        "80EC0BCD-D279-DF11-B1DB-0030487C90EE.root";
+
+    CHECK("/FULL_PATH_TO_THE_FIRST_STEP_ROOT_FILE/80EC0BCD-D279-DF11-B1DB-0030487C90EE.root" == fl.pfn(overriden_file));
+
+    for (auto f : lfn) {
+      CHECK("" == fl.pfn(f));
     }
-
-    {
-      std::string CMSSW_BASE(std::getenv("CMSSW_BASE"));
-      std::string CMSSW_RELEASE_BASE(std::getenv("CMSSW_RELEASE_BASE"));
-      std::string file_name("/src/FWCore/Catalog/test/override_catalog.xml");
-      std::string full_file_name = boost::filesystem::exists((CMSSW_BASE + file_name).c_str())
-                                       ? CMSSW_BASE + file_name
-                                       : CMSSW_RELEASE_BASE + file_name;
-
-      edm::FileLocator fl(("trivialcatalog_file:" + full_file_name + "?protocol=override").c_str(), false);
-
-      const char* lfn[] = {
-          "/store/unmerged/relval/CMSSW_3_8_0_pre3/RelValZTT/GEN-SIM-DIGI-RAW-HLTDEBUG/START38_V2-v1/0666/"
-          "80EC0BCD-D279-DF11-B1DB-0030487C90EE.root",
-          "/store/group/bha/bho",
-          "/bha/bho",
-          "bha",
-          "file:bha",
-          "file:/bha/bho",
-          "/castor/cern.ch/cms/bha/bho",
-          "rfio:/castor/cern.ch/cms/bha/bho",
-          "rfio:/bha/bho"};
-      int nfile = 9;
-
-      std::cout << "lfn2pfn" << std::endl;
-      for (int i = 0; i < nfile; ++i)
-        std::cout << lfn[i] << " -> " << fl.pfn(lfn[i]) << std::endl;
-
-      std::cout << "pfn2lfn" << std::endl;
-      for (int i = 0; i < nfile; ++i)
-        std::cout << lfn[i] << " -> " << fl.lfn(lfn[i]) << std::endl;
+    for (auto f : lfn) {
+      CHECK("" == fl.lfn(f));
     }
-
-  } catch (cms::Exception const& e) {
-    std::cout << e.what() << std::endl;
-    return 1;
   }
-
-  catch (...) {
-    std::cout << "got a problem..." << std::endl;
-    return 1;
-  }
-
-  return 0;
 }

--- a/FWCore/Catalog/test/simple_catalog.xml
+++ b/FWCore/Catalog/test/simple_catalog.xml
@@ -1,0 +1,7 @@
+<storage-mapping>
+  <lfn-to-pfn protocol="xrd"
+  path-match="/+store/(.*)" result="/storage/path/store/$1"/>
+  <pfn-to-lfn protocol="xrd"
+  path-match="^/+storage/path/store/(.*)" result="/store/$1"/>
+</storage-mapping>
+

--- a/FWCore/Services/BuildFile.xml
+++ b/FWCore/Services/BuildFile.xml
@@ -8,11 +8,10 @@
 <use   name="FWCore/Utilities"/>
 <use   name="FWCore/Concurrency"/>
 <use   name="FWCore/Framework"/>
-<use   name="Utilities/Xerces"/>
 <use   name="boost"/>
+<use   name="tinyxml2"/>
 <use   name="rootcore"/>
 <use   name="roothistmatrix"/>
-<use   name="xerces-c"/>
 <export>
   <lib   name="1"/>
 </export>

--- a/FWCore/Services/src/SiteLocalConfigService.cc
+++ b/FWCore/Services/src/SiteLocalConfigService.cc
@@ -5,16 +5,10 @@
 #include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 #include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
-#include <xercesc/dom/DOM.hpp>
-#include <xercesc/parsers/XercesDOMParser.hpp>
-#include "Utilities/Xerces/interface/Xerces.h"
-#include "Utilities/Xerces/interface/XercesStrUtils.h"
+#include "tinyxml2.h"
 #include <sstream>
 #include <memory>
 #include <boost/algorithm/string.hpp>
-
-using namespace xercesc;
-using namespace cms::xerces;
 
 //<<<<<< PRIVATE DEFINES                                                >>>>>>
 //<<<<<< PRIVATE CONSTANTS                                              >>>>>>
@@ -30,30 +24,12 @@ namespace {
 
   // concatenate all the XML node attribute/value pairs into a
   // paren-separated string (for use by CORAL and frontier_client)
-  inline std::string _toParenString(DOMNode const &nodeToConvert) {
+  inline std::string _toParenString(tinyxml2::XMLElement const &nodeToConvert) {
     std::ostringstream oss;
 
-    DOMNodeList *childList = nodeToConvert.getChildNodes();
-
-    XMLSize_t numNodes = childList->getLength();
-    for (XMLSize_t i = 0; i < numNodes; ++i) {
-      DOMNode *childNode = childList->item(i);
-      if (childNode->getNodeType() != DOMNode::ELEMENT_NODE) {
-        continue;
-      }
-      DOMElement *child = static_cast<DOMElement *>(childNode);
-
-      DOMNamedNodeMap *attributes = child->getAttributes();
-      XMLSize_t numAttributes = attributes->getLength();
-      for (XMLSize_t j = 0; j < numAttributes; ++j) {
-        DOMNode *attributeNode = attributes->item(j);
-        if (attributeNode->getNodeType() != DOMNode::ATTRIBUTE_NODE) {
-          continue;
-        }
-        DOMAttr *attribute = static_cast<DOMAttr *>(attributeNode);
-
-        oss << "(" << toString(child->getTagName()) << toString(attribute->getName()) << "="
-            << toString(attribute->getValue()) << ")";
+    for (auto child = nodeToConvert.FirstChildElement(); child != nullptr; child = child->NextSiblingElement()) {
+      for (auto attribute = child->FirstAttribute(); attribute != nullptr; attribute = attribute->Next()) {
+        oss << "(" << child->Name() << attribute->Name() << "=" << attribute->Value() << ")";
       }
     }
     return oss.str();
@@ -66,6 +42,15 @@ namespace {
       iPointer = &iHolder;
     }
   }
+
+  constexpr char const *const kEmptyString = "";
+  const char *safe(const char *iCheck) {
+    if (iCheck == nullptr) {
+      return kEmptyString;
+    }
+    return iCheck;
+  }
+
 }  // namespace
 
 namespace edm {
@@ -74,7 +59,8 @@ namespace edm {
     const std::string SiteLocalConfigService::m_statisticsDefaultPort = "3334";
 
     SiteLocalConfigService::SiteLocalConfigService(ParameterSet const &pset)
-        : m_url("/SITECONF/local/JobConfig/site-local-config.xml"),
+        : m_url(pset.getUntrackedParameter<std::string>("siteLocalConfigFileUrl",
+                                                        "/SITECONF/local/JobConfig/site-local-config.xml")),
           m_dataCatalog(),
           m_fallbackDataCatalog(),
           m_frontierConnect(),
@@ -103,10 +89,12 @@ namespace edm {
           m_statisticsAddrInfo(nullptr),
           m_statisticsInfoAvail(false),
           m_siteName() {
-      char *tmp = getenv("CMS_PATH");
-
-      if (tmp) {
-        m_url = tmp + m_url;
+      if (m_url.empty()) {
+        char *tmp = getenv("CMS_PATH");
+        m_url = "/SITECONF/local/JobConfig/site-local-config.xml";
+        if (tmp) {
+          m_url = tmp + m_url;
+        }
       }
 
       this->parse(m_url);
@@ -278,206 +266,162 @@ namespace edm {
     std::string const &SiteLocalConfigService::siteName() const { return m_siteName; }
 
     void SiteLocalConfigService::parse(std::string const &url) {
-      cms::concurrency::xercesInitialize();
-      {
-        auto parser = std::make_unique<XercesDOMParser>();
-        try {
-          parser->setValidationScheme(XercesDOMParser::Val_Auto);
-          parser->setDoNamespaces(false);
+      tinyxml2::XMLDocument doc;
+      auto loadErr = doc.LoadFile(url.c_str());
+      if (loadErr != tinyxml2::XML_SUCCESS) {
+        return;
+      }
 
-          parser->parse(url.c_str());
-          DOMDocument *doc = parser->getDocument();
-          if (!doc) {
-            return;
-          }
+      // The Site Config has the following format
+      // <site-local-config>
+      // <site name="FNAL">
+      //   <event-data>
+      //     <catalog url="trivialcatalog_file:/x/y/z.xml"/>
+      //     <rfiotype value="castor"/>
+      //   </event-data>
+      //   <calib-data>
+      //     <catalog url="trivialcatalog_file:/x/y/z.xml"/>
+      //     <frontier-connect>
+      //       ... frontier-interpreted server/proxy xml ...
+      //     </frontier-connect>
+      //   </calib-data>
+      //   <source-config>
+      //     <cache-temp-dir name="/a/b/c"/>
+      //     <cache-hint value="..."/>
+      //     <read-hint value="..."/>
+      //     <ttree-cache-size value="0"/>
+      //     <native-protocols>
+      //        <protocol  prefix="dcache"/>
+      //        <protocol prefix="file"/>
+      //     </native-protocols>
+      //   </source-config>
+      // </site>
+      // </site-local-config>
+      auto rootElement = doc.RootElement();
 
-          // The Site Config has the following format
-          // <site-local-config>
-          // <site name="FNAL">
-          //   <event-data>
-          //     <catalog url="trivialcatalog_file:/x/y/z.xml"/>
-          //     <rfiotype value="castor"/>
-          //   </event-data>
-          //   <calib-data>
-          //     <catalog url="trivialcatalog_file:/x/y/z.xml"/>
-          //     <frontier-connect>
-          //       ... frontier-interpreted server/proxy xml ...
-          //     </frontier-connect>
-          //   </calib-data>
-          //   <source-config>
-          //     <cache-temp-dir name="/a/b/c"/>
-          //     <cache-hint value="..."/>
-          //     <read-hint value="..."/>
-          //     <ttree-cache-size value="0"/>
-          //     <native-protocols>
-          //        <protocol  prefix="dcache"/>
-          //        <protocol prefix="file"/>
-          //     </native-protocols>
-          //   </source-config>
-          // </site>
-          // </site-local-config>
+      for (auto site = rootElement->FirstChildElement("site"); site != nullptr;
+           site = site->NextSiblingElement("site")) {
+        // Parse the site name
+        m_siteName = safe(site->Attribute("name"));
 
-          // FIXME: should probably use the parser for validating the XML.
-
-          DOMNodeList *sites = doc->getElementsByTagName(uStr("site").ptr());
-          XMLSize_t numSites = sites->getLength();
-          for (XMLSize_t i = 0; i < numSites; ++i) {
-            DOMElement *site = static_cast<DOMElement *>(sites->item(i));
-
-            // Parse the site name
-            m_siteName = toString(site->getAttribute(uStr("name").ptr()));
-
-            // Parsing of the event data section
-            {
-              DOMNodeList *eventDataList = site->getElementsByTagName(uStr("event-data").ptr());
-              if (eventDataList->getLength() > 0) {
-                DOMElement *eventData = static_cast<DOMElement *>(eventDataList->item(0));
-
-                DOMNodeList *catalogs = eventData->getElementsByTagName(uStr("catalog").ptr());
-
-                if (catalogs->getLength() > 0) {
-                  DOMElement *catalog = static_cast<DOMElement *>(catalogs->item(0));
-                  m_dataCatalog = toString(catalog->getAttribute(uStr("url").ptr()));
-                }
-
-                if (catalogs->getLength() > 1) {
-                  DOMElement *catalog = static_cast<DOMElement *>(catalogs->item(1));
-                  m_fallbackDataCatalog = toString(catalog->getAttribute(uStr("url").ptr()));
-                }
-
-                DOMNodeList *rfiotypes = eventData->getElementsByTagName(uStr("rfiotype").ptr());
-
-                if (rfiotypes->getLength() > 0) {
-                  DOMElement *rfiotype = static_cast<DOMElement *>(rfiotypes->item(0));
-                  m_rfioType = toString(rfiotype->getAttribute(uStr("value").ptr()));
-                }
+        // Parsing of the event data section
+        {
+          auto eventData = site->FirstChildElement("event-data");
+          if (eventData) {
+            auto catalog = eventData->FirstChildElement("catalog");
+            if (catalog) {
+              m_dataCatalog = safe(catalog->Attribute("url"));
+              catalog = catalog->NextSiblingElement("catalog");
+              if (catalog) {
+                m_fallbackDataCatalog = safe(catalog->Attribute("url"));
               }
             }
-
-            // Parsing of the calib-data section
-            {
-              DOMNodeList *calibDataList = site->getElementsByTagName(uStr("calib-data").ptr());
-
-              if (calibDataList->getLength() > 0) {
-                DOMElement *calibData = static_cast<DOMElement *>(calibDataList->item(0));
-                DOMNodeList *frontierConnectList = calibData->getElementsByTagName(uStr("frontier-connect").ptr());
-
-                if (frontierConnectList->getLength() > 0) {
-                  DOMElement *frontierConnect = static_cast<DOMElement *>(frontierConnectList->item(0));
-                  m_frontierConnect = _toParenString(*frontierConnect);
-                }
-              }
-            }
-            // Parsing of the source config section
-            {
-              DOMNodeList *sourceConfigList = site->getElementsByTagName(uStr("source-config").ptr());
-
-              if (sourceConfigList->getLength() > 0) {
-                DOMElement *sourceConfig = static_cast<DOMElement *>(sourceConfigList->item(0));
-                DOMNodeList *cacheTempDirList = sourceConfig->getElementsByTagName(uStr("cache-temp-dir").ptr());
-
-                if (cacheTempDirList->getLength() > 0) {
-                  DOMElement *cacheTempDir = static_cast<DOMElement *>(cacheTempDirList->item(0));
-                  m_cacheTempDir = toString(cacheTempDir->getAttribute(uStr("name").ptr()));
-                  m_cacheTempDirPtr = &m_cacheTempDir;
-                }
-
-                DOMNodeList *cacheMinFreeList = sourceConfig->getElementsByTagName(uStr("cache-min-free").ptr());
-
-                if (cacheMinFreeList->getLength() > 0) {
-                  DOMElement *cacheMinFree = static_cast<DOMElement *>(cacheMinFreeList->item(0));
-                  m_cacheMinFree = toDouble(cacheMinFree->getAttribute(uStr("value").ptr()));
-                  m_cacheMinFreePtr = &m_cacheMinFree;
-                }
-
-                DOMNodeList *cacheHintList = sourceConfig->getElementsByTagName(uStr("cache-hint").ptr());
-
-                if (cacheHintList->getLength() > 0) {
-                  DOMElement *cacheHint = static_cast<DOMElement *>(cacheHintList->item(0));
-                  m_cacheHint = toString(cacheHint->getAttribute(uStr("value").ptr()));
-                  m_cacheHintPtr = &m_cacheHint;
-                }
-
-                DOMNodeList *cloneCacheHintList = sourceConfig->getElementsByTagName(uStr("clone-cache-hint").ptr());
-
-                if (cloneCacheHintList->getLength() > 0) {
-                  DOMElement *cloneCacheHint = static_cast<DOMElement *>(cloneCacheHintList->item(0));
-                  m_cloneCacheHint = toString(cloneCacheHint->getAttribute(uStr("value").ptr()));
-                  m_cloneCacheHintPtr = &m_cloneCacheHint;
-                }
-
-                DOMNodeList *readHintList = sourceConfig->getElementsByTagName(uStr("read-hint").ptr());
-
-                if (readHintList->getLength() > 0) {
-                  DOMElement *readHint = static_cast<DOMElement *>(readHintList->item(0));
-                  m_readHint = toString(readHint->getAttribute(uStr("value").ptr()));
-                  m_readHintPtr = &m_readHint;
-                }
-
-                DOMNodeList *ttreeCacheSizeList = sourceConfig->getElementsByTagName(uStr("ttree-cache-size").ptr());
-
-                if (ttreeCacheSizeList->getLength() > 0) {
-                  DOMElement *ttreeCacheSize = static_cast<DOMElement *>(ttreeCacheSizeList->item(0));
-                  m_ttreeCacheSize = toUInt(ttreeCacheSize->getAttribute(uStr("value").ptr()));
-                  m_ttreeCacheSizePtr = &m_ttreeCacheSize;
-                }
-
-                DOMNodeList *timeoutList = sourceConfig->getElementsByTagName(uStr("timeout-in-seconds").ptr());
-
-                if (timeoutList->getLength() > 0) {
-                  DOMElement *timeout = static_cast<DOMElement *>(timeoutList->item(0));
-                  m_timeout = toUInt(timeout->getAttribute(uStr("value").ptr()));
-                  m_timeoutPtr = &m_timeout;
-                }
-
-                DOMNodeList *statsDestList = sourceConfig->getElementsByTagName(uStr("statistics-destination").ptr());
-
-                if (statsDestList->getLength() > 0) {
-                  DOMElement *statsDest = static_cast<DOMElement *>(statsDestList->item(0));
-                  m_statisticsDestination = toString(statsDest->getAttribute(uStr("endpoint").ptr()));
-                  if (m_statisticsDestination.empty()) {
-                    m_statisticsDestination = toString(statsDest->getAttribute(uStr("name").ptr()));
-                  }
-                  std::string tmpStatisticsInfo = toString(statsDest->getAttribute(uStr("info").ptr()));
-                  boost::split(m_statisticsInfo, tmpStatisticsInfo, boost::is_any_of("\t ,"));
-                  m_statisticsInfoAvail = !tmpStatisticsInfo.empty();
-                }
-
-                DOMNodeList *prefetchingList = sourceConfig->getElementsByTagName(uStr("prefetching").ptr());
-
-                if (prefetchingList->getLength() > 0) {
-                  DOMElement *prefetching = static_cast<DOMElement *>(prefetchingList->item(0));
-                  m_enablePrefetching = toBool(prefetching->getAttribute(uStr("value").ptr()));
-                  m_enablePrefetchingPtr = &m_enablePrefetching;
-                }
-
-                DOMNodeList *nativeProtocolsList = sourceConfig->getElementsByTagName(uStr("native-protocols").ptr());
-
-                if (nativeProtocolsList->getLength() > 0) {
-                  DOMElement *nativeProtocol = static_cast<DOMElement *>(nativeProtocolsList->item(0));
-                  DOMNodeList *childList = nativeProtocol->getChildNodes();
-
-                  XMLSize_t numNodes = childList->getLength();
-                  for (XMLSize_t i = 0; i < numNodes; ++i) {
-                    DOMNode *childNode = childList->item(i);
-                    if (childNode->getNodeType() != DOMNode::ELEMENT_NODE) {
-                      continue;
-                    }
-                    DOMElement *child = static_cast<DOMElement *>(childNode);
-                    m_nativeProtocols.push_back(toString(child->getAttribute(uStr("prefix").ptr())));
-                  }
-                  m_nativeProtocolsPtr = &m_nativeProtocols;
-                }
-              }
+            auto rfiotype = eventData->FirstChildElement("rfiotype");
+            if (rfiotype) {
+              m_rfioType = safe(rfiotype->Attribute("value"));
             }
           }
-          m_connected = true;
-        } catch (xercesc::DOMException const &e) {
         }
-      }  // The extra pair of braces ensures that
-      // all implicit destructors are called
-      // *before* terminating Xerces-C++.
-      cms::concurrency::xercesTerminate();
+
+        // Parsing of the calib-data section
+        {
+          auto calibData = site->FirstChildElement("calib-data");
+
+          if (calibData) {
+            auto frontierConnect = calibData->FirstChildElement("frontier-connect");
+
+            if (frontierConnect) {
+              m_frontierConnect = _toParenString(*frontierConnect);
+            }
+          }
+        }
+        // Parsing of the source config section
+        {
+          auto sourceConfig = site->FirstChildElement("source-config");
+
+          if (sourceConfig) {
+            auto cacheTempDir = sourceConfig->FirstChildElement("cache-temp-dir");
+
+            if (cacheTempDir) {
+              m_cacheTempDir = safe(cacheTempDir->Attribute("name"));
+              m_cacheTempDirPtr = &m_cacheTempDir;
+            }
+
+            auto cacheMinFree = sourceConfig->FirstChildElement("cache-min-free");
+
+            if (cacheMinFree) {
+              //TODO what did xerces do if it couldn't convert?
+              m_cacheMinFree = cacheMinFree->DoubleAttribute("value");
+              m_cacheMinFreePtr = &m_cacheMinFree;
+            }
+
+            auto cacheHint = sourceConfig->FirstChildElement("cache-hint");
+
+            if (cacheHint) {
+              m_cacheHint = safe(cacheHint->Attribute("value"));
+              m_cacheHintPtr = &m_cacheHint;
+            }
+
+            auto cloneCacheHint = sourceConfig->FirstChildElement("clone-cache-hint");
+
+            if (cloneCacheHint) {
+              m_cloneCacheHint = safe(cloneCacheHint->Attribute("value"));
+              m_cloneCacheHintPtr = &m_cloneCacheHint;
+            }
+
+            auto readHint = sourceConfig->FirstChildElement("read-hint");
+
+            if (readHint) {
+              m_readHint = safe(readHint->Attribute("value"));
+              m_readHintPtr = &m_readHint;
+            }
+
+            auto ttreeCacheSize = sourceConfig->FirstChildElement("ttree-cache-size");
+
+            if (ttreeCacheSize) {
+              m_ttreeCacheSize = ttreeCacheSize->UnsignedAttribute("value");
+              m_ttreeCacheSizePtr = &m_ttreeCacheSize;
+            }
+
+            auto timeout = sourceConfig->FirstChildElement("timeout-in-seconds");
+
+            if (timeout) {
+              m_timeout = timeout->UnsignedAttribute("value");
+              m_timeoutPtr = &m_timeout;
+            }
+
+            auto statsDest = sourceConfig->FirstChildElement("statistics-destination");
+
+            if (statsDest) {
+              m_statisticsDestination = safe(statsDest->Attribute("endpoint"));
+              if (m_statisticsDestination.empty()) {
+                m_statisticsDestination = safe(statsDest->Attribute("name"));
+              }
+              std::string tmpStatisticsInfo = safe(statsDest->Attribute("info"));
+              boost::split(m_statisticsInfo, tmpStatisticsInfo, boost::is_any_of("\t ,"));
+              m_statisticsInfoAvail = !tmpStatisticsInfo.empty();
+            }
+
+            auto prefetching = sourceConfig->FirstChildElement("prefetching");
+
+            if (prefetching) {
+              m_enablePrefetching = prefetching->BoolAttribute("value");
+              m_enablePrefetchingPtr = &m_enablePrefetching;
+            }
+
+            auto nativeProtocol = sourceConfig->FirstChildElement("native-protocols");
+
+            if (nativeProtocol) {
+              for (auto child = nativeProtocol->FirstChildElement(); child != nullptr;
+                   child = child->NextSiblingElement()) {
+                m_nativeProtocols.push_back(safe(child->Attribute("prefix")));
+              }
+              m_nativeProtocolsPtr = &m_nativeProtocols;
+            }
+          }
+        }
+      }
+      m_connected = true;
     }
 
     void SiteLocalConfigService::computeStatisticsDestination() {
@@ -503,6 +447,9 @@ namespace edm {
       ParameterSetDescription desc;
       desc.setComment("Service to translate logical file names to physical file names.");
 
+      desc.addOptionalUntracked<std::string>("siteLocalConfigFileUrl", std::string())
+          ->setComment(
+              "Specify the file containing the site local config. Empty string will load from default directory.");
       desc.addOptionalUntracked<std::string>("overrideSourceCacheTempDir");
       desc.addOptionalUntracked<double>("overrideSourceCacheMinFree");
       desc.addOptionalUntracked<std::string>("overrideSourceCacheHintDir");

--- a/FWCore/Services/src/SiteLocalConfigService.cc
+++ b/FWCore/Services/src/SiteLocalConfigService.cc
@@ -51,6 +51,16 @@ namespace {
     return iCheck;
   }
 
+  std::string defaultURL() {
+    std::string returnValue;
+    const char *tmp = getenv("CMS_PATH");
+    if (tmp) {
+      returnValue = tmp;
+    }
+    returnValue += "/SITECONF/local/JobConfig/site-local-config.xml";
+    return returnValue;
+  }
+
 }  // namespace
 
 namespace edm {
@@ -59,8 +69,7 @@ namespace edm {
     const std::string SiteLocalConfigService::m_statisticsDefaultPort = "3334";
 
     SiteLocalConfigService::SiteLocalConfigService(ParameterSet const &pset)
-        : m_url(pset.getUntrackedParameter<std::string>("siteLocalConfigFileUrl",
-                                                        "/SITECONF/local/JobConfig/site-local-config.xml")),
+        : m_url(pset.getUntrackedParameter<std::string>("siteLocalConfigFileUrl", defaultURL())),
           m_dataCatalog(),
           m_fallbackDataCatalog(),
           m_frontierConnect(),
@@ -89,14 +98,6 @@ namespace edm {
           m_statisticsAddrInfo(nullptr),
           m_statisticsInfoAvail(false),
           m_siteName() {
-      if (m_url.empty()) {
-        char *tmp = getenv("CMS_PATH");
-        m_url = "/SITECONF/local/JobConfig/site-local-config.xml";
-        if (tmp) {
-          m_url = tmp + m_url;
-        }
-      }
-
       this->parse(m_url);
 
       //apply overrides

--- a/FWCore/Services/test/BuildFile.xml
+++ b/FWCore/Services/test/BuildFile.xml
@@ -13,3 +13,7 @@
   <flags   TEST_RUNNER_ARGS=" /bin/bash FWCore/Services/test test_mallocopts.sh test_sitelocalconfig.sh test_resource.sh test_zombiekiller.sh"/>
   <use   name="FWCore/Utilities"/>
 </bin>
+<bin    file="test_catch2_*.cc" name="testFWCoreServicesCatch2">
+  <use name="catch2"/>
+  <use name="FWCore/Services"/>
+</bin>

--- a/FWCore/Services/test/full-site-local-config.testfile
+++ b/FWCore/Services/test/full-site-local-config.testfile
@@ -1,0 +1,31 @@
+<site-local-config>
+<site name="DUMMY">
+  <event-data>
+    <catalog url="trivialcatalog_file:/dummy/storage.xml?protocol=dcap"/>
+  </event-data>
+  <local-stage-out>
+    <se-name value="cmssrm.dummy.foo"/>
+    <command value="srm"/>
+    <catalog url="trivialcatalog_file:/dummy/storage.xml?protocol=srm"/>
+  </local-stage-out>
+  <calib-data>
+    <frontier-connect>
+      <proxy url="http://cmsfrontier.dummy.foo:3128"/>
+      <proxy url="http://cmsfrontier.dummy.foo:3128"/>
+    </frontier-connect>
+  </calib-data>
+  <source-config>
+    <cache-temp-dir name="/a/b/c"/>
+    <cache-hint value="application-only"/>
+    <read-hint value="read-ahead-buffered"/>
+    <ttree-cache-size value="10000"/>
+    <native-protocols>
+      <protocol  prefix="dcache"/>
+      <protocol prefix="file"/>
+    </native-protocols>
+    <prefetching value="true"/>
+    <statistics-destination name="cms-udpmon-collector.cern.ch:9331" info ="dn"/>
+    <timeout-in-seconds value="7200" />
+  </source-config>
+</site>
+</site-local-config>

--- a/FWCore/Services/test/test_catch2_SiteLocalConfigService.cc
+++ b/FWCore/Services/test/test_catch2_SiteLocalConfigService.cc
@@ -1,0 +1,96 @@
+#include "FWCore/Services/src/SiteLocalConfigService.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+
+TEST_CASE("Test SiteLocalConfigService", "[sitelocalconfig]") {
+  std::string dirString;
+  auto dir = getenv("LOCAL_TEST_DIR");
+  if (dir) {
+    dirString = dir;
+  } else {
+    auto dir = getenv("CMSSW_BASE");
+    if (dir) {
+      dirString = dir;
+      dirString += "/src/FWCore/Services/test";
+    }
+  }
+  REQUIRE(not dirString.empty());
+  SECTION("full-site-local-config.testfile") {
+    edm::ParameterSet pset;
+    pset.addUntrackedParameter<std::string>("siteLocalConfigFileUrl", dirString + "/full-site-local-config.testfile");
+
+    edm::service::SiteLocalConfigService slc(pset);
+
+    CHECK(slc.dataCatalog() == "trivialcatalog_file:/dummy/storage.xml?protocol=dcap");
+    CHECK(slc.fallbackDataCatalog().empty());
+    REQUIRE(slc.sourceCacheTempDir() != nullptr);
+    CHECK(*slc.sourceCacheTempDir() == "/a/b/c");
+    CHECK(slc.sourceCacheMinFree() == nullptr);
+    REQUIRE(slc.sourceCacheHint() != nullptr);
+    CHECK(*slc.sourceCacheHint() == "application-only");
+    REQUIRE(slc.sourceCloneCacheHint() == nullptr);
+    REQUIRE(slc.sourceReadHint() != nullptr);
+    CHECK(*slc.sourceReadHint() == "read-ahead-buffered");
+    REQUIRE(slc.sourceTTreeCacheSize() != nullptr);
+    CHECK(*slc.sourceTTreeCacheSize() == 10000);
+    REQUIRE(slc.sourceTimeout() != nullptr);
+    CHECK(*slc.sourceTimeout() == 7200);
+    CHECK(slc.enablePrefetching());
+    REQUIRE(slc.sourceNativeProtocols() != nullptr);
+    REQUIRE(slc.sourceNativeProtocols()->size() == 2);
+    CHECK((*slc.sourceNativeProtocols())[0] == "dcache");
+    CHECK((*slc.sourceNativeProtocols())[1] == "file");
+    REQUIRE(slc.statisticsDestination() != nullptr);
+    REQUIRE(slc.statisticsInfo() != nullptr);
+    CHECK(slc.statisticsInfo()->size() == 1);
+    CHECK(slc.statisticsInfo()->find("dn") != slc.statisticsInfo()->end());
+    CHECK(slc.siteName() == "DUMMY");
+  }
+
+  SECTION("overrides") {
+    edm::ParameterSet pset;
+    pset.addUntrackedParameter<std::string>("siteLocalConfigFileUrl", dirString + "/full-site-local-config.testfile");
+
+    pset.addUntrackedParameter<std::string>("overrideSourceCacheTempDir", "/a/d");
+    pset.addUntrackedParameter<double>("overrideSourceCacheMinFree", 10.);
+    pset.addUntrackedParameter<std::string>("overrideSourceCacheHintDir", "dummy");
+    pset.addUntrackedParameter<std::string>("overrideSourceCloneCacheHintDir", "foo");
+    pset.addUntrackedParameter<std::string>("overrideSourceReadHint", "read-ahead-unbuffered");
+    pset.addUntrackedParameter<std::vector<std::string> >("overrideSourceNativeProtocols", {{"http"}});
+    pset.addUntrackedParameter<unsigned int>("overrideSourceTTreeCacheSize", 20000);
+    pset.addUntrackedParameter<unsigned int>("overrideSourceTimeout", 0);
+    pset.addUntrackedParameter<bool>("overridePrefetching", false);
+    pset.addUntrackedParameter<std::string>("overrideStatisticsDestination", "");
+    pset.addUntrackedParameter<std::vector<std::string> >("overrideStatisticsInfo", {{"nodn"}});
+
+    edm::service::SiteLocalConfigService slc(pset);
+
+    CHECK(slc.dataCatalog() == "trivialcatalog_file:/dummy/storage.xml?protocol=dcap");
+    CHECK(slc.fallbackDataCatalog().empty());
+    REQUIRE(slc.sourceCacheTempDir() != nullptr);
+    CHECK(*slc.sourceCacheTempDir() == "/a/d");
+    REQUIRE(slc.sourceCacheMinFree() != nullptr);
+    CHECK(*slc.sourceCacheMinFree() == 10.);
+    REQUIRE(slc.sourceCacheHint() != nullptr);
+    CHECK(*slc.sourceCacheHint() == "dummy");
+    REQUIRE(slc.sourceCloneCacheHint() != nullptr);
+    CHECK(*slc.sourceCloneCacheHint() == "foo");
+    REQUIRE(slc.sourceReadHint() != nullptr);
+    CHECK(*slc.sourceReadHint() == "read-ahead-unbuffered");
+    REQUIRE(slc.sourceTTreeCacheSize() != nullptr);
+    CHECK(*slc.sourceTTreeCacheSize() == 20000);
+    REQUIRE(slc.sourceTimeout() != nullptr);
+    CHECK(*slc.sourceTimeout() == 0);
+    CHECK(not slc.enablePrefetching());
+    REQUIRE(slc.sourceNativeProtocols() != nullptr);
+    REQUIRE(slc.sourceNativeProtocols()->size() == 1);
+    CHECK((*slc.sourceNativeProtocols())[0] == "http");
+    REQUIRE(slc.statisticsDestination() == nullptr);
+    REQUIRE(slc.statisticsInfo() != nullptr);
+    CHECK(slc.statisticsInfo()->size() == 1);
+    CHECK(slc.statisticsInfo()->find("nodn") != slc.statisticsInfo()->end());
+    CHECK(slc.siteName() == "DUMMY");
+  }
+}

--- a/L1TriggerConfig/L1GtConfigProducers/BuildFile.xml
+++ b/L1TriggerConfig/L1GtConfigProducers/BuildFile.xml
@@ -5,6 +5,7 @@
 <use   name="CondFormats/DataRecord"/>
 <use   name="CondTools/L1Trigger"/>
 <use   name="HLTrigger/HLTcore"/>
+<use   name="Utilities/Xerces"/>
 <use   name="xerces-c"/>
 <export>
   <lib   name="1"/>

